pkg/asset/machines/vsphere/machines.go

```diff
@@ -24,41 +21,36 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 		return nil, fmt.Errorf("non-VSphere machine-pool: %q", poolPlatform)
 	}
 
+	var failureDomain vsphere.FailureDomain
 	var machines []machineapi.Machine
-
 	platform := config.Platform.VSphere
 	mpool := pool.Platform.VSphere
+	replicas := int64(1)
+	definedZones := make(map[string]vsphere.FailureDomain)
 
-	azs := mpool.Zones
-	numOfAZs := len(azs)
-	definedZones := make(map[string]*vsphere.Platform)
+	numOfZones := len(mpool.Zones)
 
-	if numOfAZs > 0 {
-		zones, err := getDefinedZones(platform)
+	if numOfZones > 0 {
```

Comment:
Isn't this always at least 1?

Reply:
yeah you're right, there is no reason to check for that anymore